src/debug.c

```diff
@@ -1018,34 +1018,7 @@ NULL
             return;
         }
         addReply(c, shared.ok);
-    } else if(!strcasecmp(c->argv[1]->ptr,"CLUSTERLINK") &&
-        !strcasecmp(c->argv[2]->ptr,"KILL") &&
-        c->argc == 5) {
-        if (!server.cluster_enabled) {
-            addReplyError(c, "Debug option only available for cluster mode enabled setup!");
-            return;
-        }
-
-        /* Find the node. */
-        clusterNode *n = clusterLookupNode(c->argv[4]->ptr, sdslen(c->argv[4]->ptr));
-        if (!n) {
-            addReplyErrorFormat(c,"Unknown node %s", (char*)c->argv[4]->ptr);
-            return;
-        }
-
-        /* Terminate the link based on the direction or all. */
-        if (!strcasecmp(c->argv[3]->ptr,"from")) {
-            freeClusterLink(n->inbound_link);
-        } else if (!strcasecmp(c->argv[3]->ptr,"to")) {
-            freeClusterLink(n->link);
-        } else if (!strcasecmp(c->argv[3]->ptr,"all")) {
-            freeClusterLink(n->link);
-            freeClusterLink(n->inbound_link);
-        } else {
-            addReplyErrorFormat(c, "Unknown direction %s", (char*) c->argv[3]->ptr);
-        }
-        addReply(c,shared.ok);
-    } else {
+    } else if(!handleDebugClusterCommand(c)) {
```

Comment:
maybe this should be a new sub-command in CLUSTER?
like we have with SENTINEL DEBUG?

Reply:
note that DEBUG HELP still refers to CLUSTERLINK KILL

Reply:
@oranagra If we want to move this to a different command I propose that we do that in a separate PR, OK? I do not understand your second comment.

Reply:
Yes. Let's keep the interface changes for later.
The thing is that debug.c has the list if sub-commands in DEBUG HELP.
but each implementation of cluster.c implements it separately, and can support a different set of sub-commands.
Maybe we need to split the code in a way that the c files can only implement one specific sub-command. Or move the portion of the help message to a separate cluster function too

Reply:
OMG, I forgot to merge that. Will fix in a way similar to how I handled cluster command help. That was in the original PR and I forgot

Reply:
I thought about this some more and decided that moving forward, especially for flotilla, it makes sense to have a cluster subcommand "debug" that can do all kinds of stuff (we have all kinds of plans already). As such, I took a minimalist approach to this function and simply put an #ifndef around CLUSTERLINK KILL. @oranagra, OK if I mark this resolved?

Reply:
It's still odd that the sub-command list is documented in one file, but the string matching for that sub-command is in another.
IMHO we have 3 options:
1. Make a dedicated function for node killing, have the string matching done in debug.c
2. Add a function that returns the help string of the cluster related sub-commands, so that each implementation can document and implement any sub-command list it wishes.
3. Introduce a sub-sub-command named DEBUG CLUSTER or a command named CLUSTER DEBUG. (the first option is preferable since it doesn't add a new command). Note that changes inside the DEBUG command are not considered breaking changes (only meant to be used by tests)

I prefer option 2.  WDYT?

Reply:
I am fine with option 2, that's what I originally had in a previous version. Will do that Monday AM first thing.

Reply:
done